pika/heartbeat.py

```diff
@@ -9,37 +9,32 @@
 
 class HeartbeatChecker(object):
     """Checks to make sure that our heartbeat is received at the expected
-    intervals.
+    timeouts.
 
     """
-    DEFAULT_INTERVAL = 60
-    MAX_IDLE_COUNT = 2
-
+    # Note: even though we're sending heartbeats in half the specified
+    # timeout value, the broker will be sending them to us at the specified
+    # value. This means we'll be checking for an idle connection via send_and_check
+    # twice as many times as the broker will send heartbeats to us,
+    # so we need to set max idle count to 4 here
+    _MAX_IDLE_COUNT = 4
```

Comment:
What made you switch from 2 to 4? Not saying it's wrong, just curious?

Reply:
Here's what I observed. A 60 second negotiated timeout will result in RabbitMQ sending heartbeats every 60 seconds. Since we changed Pika to send heartbeats every `timeout / 2` seconds (to prevent edge cases that can end up with RMQ closing a connection), heartbeats will be sent to RMQ every 30 seconds. We also check for an idle connection in `send_and_check` in Pika, so we'll be checking twice as often. Once, when testing my change, the timing was just right that the idle count reached 2 and Pika erroneously thought the connection was idle. So, I think it makes sense to basically double the idle count to go with halving the heartbeat timeout.

Reply:
I will re-run my test with Wireshark, but I'm pretty sure my observation is correct when a 60 second heartbeat is used with an app that uses the Java client.